src/Network/Algorand/Client.hs

```diff
@@ -68,26 +47,34 @@ connectToNode host net = do
   manager <- newTlsManager
   env <- mkClientEnv manager <$> parseBaseUrl (T.unpack host)
   let
+    apiAny :: ApiAny (AsClientT m)
+    apiAny = fromServant $ _vAny $ apiClient env
+
     apiV2Client :: forall m' . MonadIO m' => ApiV2 (AsClientT m')
     apiV2Client = fromServant $ _v2 $ apiClient env
-  connect env net (AlgoNode apiV2Client)
+
+  version@Version{vGenesisId} <- _version apiAny
+  if vGenesisId == net
+  then pure (version, AlgoNode apiV2Client)
+  else throwM $ WrongNetwork net vGenesisId
 
 newtype AlgoIndexer = AlgoIndexer
   { getAlgoIndexer :: forall m' . MonadIO m' => ApiIdx2 (AsClientT m')
   }
 
--- | Connect to an indexer and make sure it is working on the expected network.
+-- | Connect to an indexer.
 connectToIndexer
   :: forall m. (MonadIO m, MonadThrow m)
   => Host
   -- ^ Host of the indexer.
   -> Network
   -- ^ Expected network (genesis id).
```

Comment:
Does it make sense to still pass `Network` here if we do nothing with it and just return as is?

Reply:
I didn't want to change much code and exclude network, because at node we check network version, but indexer doesn't have such endpoint to check it
and we do not no will it be or not, so let's keep it as it is for now, and later update to check network (iff there are will be such endpoint) or delete this

Reply:
Ok, let's think about it in SDAO-479. I'm ok with merging (after squashing fixups) as is, since there will be another part in SDAO-479.